### PR TITLE
[DEVX-14]FR: pass base URL

### PR DESCRIPTION
**Pull request by sanjaychelliah**
- Support passing [base URL](https://clarifai.slack.com/archives/C056AUEP60J/p1683651213066839) in data_upload

### CODE REVIEW

File: clarifai/auth/helper.py [hunk 1]
```diff
@@ -273,16 +273,7 @@ def get_stub(self) -> service_pb2_grpc.V2Stub:
     if self._base not in base_https_cache:
       raise Exception("Cannot determine if base %s is https" % self._base)
 
-    https = base_https_cache[self._base]
```
Comment (yangshengBE) [inline clarifai/auth/helper.py:276]: i think we probably need to revert this

**Review by yangshengBE — COMMENTED** [review]
I think our base auth pkg is good, no need to change but the upload.py is incompatible

File: clarifai/data_upload/upload.py [hunk 2]
```diff
@@ -92,7 +92,8 @@ def __init__(
       from_zoo: Optional[str] = None,  # load dataset from zoo
       split: str = "train",  # train or test/val
       chunk_size: int = 128,
-      portal: str = "clarifai"):
+      portal: str = None,
```
Comment (yangshengBE) [inline clarifai/data_upload/upload.py:95]: i do not think we need portal url

File: clarifai/data_upload/upload.py [hunk 3]
```diff
@@ -122,12 +123,20 @@ def __init__(
     self.annot_num_workers = 4
     self.max_retires = 10
     self.__base: str = ""
-    if portal == "dev":
-      self.__base = "https://api-dev.clarifai.com"
-    elif portal == "staging":
-      self.__base = "https://api-staging.clarifai.com"
-    else:  #prod
-      self.__base = "https://api.clarifai.com"
+    if portal and base_url:
```
Comment (yangshengBE) [inline clarifai/data_upload/upload.py:126]: we should just set self._base to whatever user provided (ok to be none)

Comment (steph-en-m) [reply clarifai/data_upload/upload.py:126]: Right, 
I think the auth should be updated too to use ClarifaiAuthHelper() and then the auth instance object passed to create_stub(): that way the DEFAULT_BASE is used as opposed to the current auth that requires all auth variables be exported as env variables.

File: clarifai/data_upload/upload.py [hunk 4]
```diff
@@ -123,28 +122,14 @@ def __init__(
...
-    os.environ["CLARIFAI_API_BASE"] = self.__base
-    os.environ["CLARIFAI_PAT"] = self.PAT
+      self.__base = base_url
 
-    self.STUB: service_pb2_grpc.V2Stub = create_stub()
+    auth_helper = ClarifaiAuthHelper(
+        user_id=self.USER_ID, app_id=self.APP_ID, pat=self.PAT, base=self.__base)
+    self.STUB: service_pb2_grpc.V2Stub = create_stub(auth_helper)
     self.metadata: Tuple = (('authorization', 'Key ' + self.PAT),)
     self.user_app_id = resources_pb2.UserAppIDSet(user_id=self.USER_ID, app_id=self.APP_ID)
```
[the remark this answers is not in the document]
Comment (sanjaychelliah) [reply clarifai/data_upload/upload.py]: Thanks for the catch @steph-en-m . Changed it now..

**Review by steph-en-m — APPROVED** [review]

**Pull request merged**